shared-brotli-patch-decoder/src/lib.rs

```diff
@@ -9,42 +9,77 @@ pub mod decode_error;
 #[cfg(fuzzing)]
 use decode_error::DecodeError;
 
-/// Decodes shared brotli encoded data using the optional shared dictionary.
-///
-/// The shared dictionary is a raw LZ77 style dictionary, see:
-/// <https://datatracker.ietf.org/doc/html/draft-vandevenne-shared-brotli-format-11#section-3.2>
-///
-/// Will fail if the decoded result will be greater then max_uncompressed_length. Any excess data
-/// in encoded after the encoded stream finishes is also considered an error.
-pub fn shared_brotli_decode(
-    encoded: &[u8],
-    shared_dictionary: Option<&[u8]>,
-    max_uncompressed_length: usize,
-) -> Result<Vec<u8>, decode_error::DecodeError> {
-    cfg_if::cfg_if! {
-        if #[cfg(fuzzing)]
-        {
-            // When running under a fuzzer disable brotli decoding and instead just pass through the input data.
-            // This allows the fuzzer to more effectively explore code gated behind brotli decoding.
-            // TODO(garretrieger): instead consider modifying the top level IFT apis to allow a custom brotli decoder
-            //   implementation to be provided. This would allow fuzzing to sub in a custom impl that could return all
-            //   of the possible errors that the standard impl here can generate.
-            return if encoded.len() <= max_uncompressed_length {
-                Ok(encoded.to_vec())
+/// A shared brotli decoder.
```

Comment:
Shared with who? I'd like a bit more info on this trait, since naively it feels like `BrotliDecoder` would be a reasonable name?

Reply:
Shared brotli (https://datatracker.ietf.org/doc/draft-vandevenne-shared-brotli-format/) is the name of a standardized extension to the original brotli specification (specifically this is what adds support for shared dictionaries)